crates/task-impls/src/da.rs

```diff
@@ -57,6 +57,9 @@ pub struct DATaskState<
     /// Membership for the DA committee
     pub da_membership: Arc<TYPES::Membership>,
 
+    /// Membership for the quorum committee
+    pub quorum_membership: Arc<TYPES::Membership>,
```

Comment:
I feel a bit counter-intuitive to store quorum membership in the DA task. If all we need is the number of quorum nodes for the VID commitment computation, could we pass this info differently?

A potential way is to add `total_nodes` to the `TransactionsSequenced` event in the VID task (which knows the quorum membership), and pass that to `DAProposalSend`, then `DAProposalRecv`, so that the DA task will have `total_nodes` without storing the quorum membership.

Reply:
Agree that `quorum_membership` should not be part of `DATaskState`.

Checking my understanding: my intuition is that each node knows the number of nodes in the quorum membership, so there should be no need to pass this info around in events. But I guess this info is only available inside a task whose state includes `quorum_membership`, and we don't want to pollute tasks with this data if they need only the node count. Correct?

In that case, yeah, I guess Keyao's solution seems best.

Reply:
I like passing around the entire scheme, I think it lends much better to future changes.

I wonder if the scheme in `DATaskState` would have to change - do we expect to have a dynamic number of storage nodes?

Reply:
Relevant: https://github.com/EspressoSystems/HotShot/pull/2163#discussion_r1419461463

Each node has all the info it needs to build the VidScheme. IMO it's merely a matter what's the cleanest way to access this info. It would suck to add the whole quorum exchange to the DA state just to get the quorum size but I don't know about other alternatives. Need some insight from seasoned veterans of the hotshot codebase.

Reply:
Sounds good!